src/components/implementation/no_interface/cFE_booter/ostask.c

```diff
@@ -0,0 +1,794 @@
+
+	struct sl_thd_policy *policy = sl_mod_thd_policy_get(thd);
+	strcpy(policy->osal_task_prop.name, task_name);
+	policy->osal_task_prop.creator    = OS_TaskGetId();
+	policy->osal_task_prop.stack_size = stack_size;
+	policy->osal_task_prop.priority   = priority;
+	policy->osal_task_prop.OStask_id  = (uint32)sl_thd_thdid(thd);
+	policy->delete_handler            = NULL;
+
+	*task_id = (uint32)sl_thd_thdid(thd);
+
+	return OS_SUCCESS;
+}
+
+int32
+OS_TaskDelete(uint32 task_id)
+{
+	struct sl_thd *thd = sl_thd_lkup(task_id);
+	if (!thd) { return OS_ERR_INVALID_ID; }
+
+	struct sl_thd_policy *thd_policy = sl_mod_thd_policy_get(thd);
+
+	osal_task_entry delete_handler = thd_policy->delete_handler;
+	if (delete_handler) { delete_handler(); }
+
+	sl_thd_free(thd);
+
+	return OS_SUCCESS;
+}
+
+uint32
+OS_TaskGetId(void)
+{
+	thdid_t real_id           = sl_thdid();
+	thdid_t possible_override = id_overrides[real_id];
+	if (possible_override) return possible_override;
+	return real_id;
+}
+
+void
+OS_TaskExit(void)
+{
+	sl_thd_free(sl_thd_curr());
+	PANIC("Should be unreachable!");
+}
+
+int32
+OS_TaskInstallDeleteHandler(osal_task_entry function_pointer)
+{
+	if (OS_TaskGetId() == main_delegate_thread_id) { return OS_ERR_INVALID_ID; }
+	sl_mod_thd_policy_get(sl_thd_curr())->delete_handler = function_pointer;
+	return OS_SUCCESS;
+}
+
+int32
+OS_TaskDelay(uint32 millisecond)
+{
+	cycles_t wakeup = sl_now() + sl_usec2cyc(millisecond * 1000);
+	sl_thd_block_timeout(0, wakeup);
+	return OS_SUCCESS;
+}
+
+int32
+OS_TaskSetPriority(uint32 task_id, uint32 new_priority)
+{
+	if (new_priority > 255 || new_priority < 1) { return OS_ERR_INVALID_PRIORITY; }
+
+	struct sl_thd *thd = sl_thd_lkup(task_id);
+	if (!thd) { return OS_ERR_INVALID_ID; }
+
+	union sched_param_union sp = {.c = {.type = SCHEDP_PRIO, .value = new_priority}};
+	sl_thd_param_set(thd, sp.v);
+
+	return OS_SUCCESS;
+}
+
+int32
+OS_TaskRegister(void)
+{
+	if (OS_TaskGetId() == main_delegate_thread_id) { return OS_ERR_INVALID_ID; }
+
+	return OS_SUCCESS;
+}
+
+
+int32
+OS_TaskGetIdByName(uint32 *task_id, const char *task_name)
+{
+	// FIXME: Implement this
+	return OS_ERR_NOT_IMPLEMENTED;
+}
+
+int32
+OS_TaskGetInfo(uint32 task_id, OS_task_prop_t *task_prop)
+{
+	if (!task_prop) { return OS_INVALID_POINTER; }
+
+	struct sl_thd *thd = sl_thd_lkup(task_id);
+	// TODO: Fix this ugly workaround
+	if (!thd || thd->state == SL_THD_FREE) { return OS_ERR_INVALID_ID; }
+	struct sl_thd_policy *thd_policy = sl_mod_thd_policy_get(thd);
+	assert(thd_policy);
+
+	*task_prop = thd_policy->osal_task_prop;
+	return OS_SUCCESS;
+}
+
+/*
+** Main thread waiting API
+*/
+
+/*
+** OS-specific background thread implementation - waits forever for events to occur.
+**
+** This should be called from the BSP main routine / initial thread after all other
+** board / application initialization has taken place and all other tasks are running.
+*/
+void
+OS_IdleLoop(void)
+{
+	while (1) sl_thd_block(0);
```

Comment:
should be `while (1) sl_thd_yield();` instead so if there is nothing else to run, this runs instead of blocking.
Anyone wakes this up?

Reply:
This is called from the root thread, which has a super high priority. Your suggestion would imply we'd need to go through a yield every time we schedule a thread. (Even though that thread would never do anything but yield!)

Reply:
My bad, I missed the comment before that function. Yeah, yield is not the right option here.